### DISCUSSION TITLE
How to sign in with oauth for current users?

### DISCUSSION THREAD

**Discussion by madhawk1296** [supabase/supabase#18064] [Questions]
I'm trying to allow users that are already signed in to connect their account to Github. But `signInWithOAuth()` removes the session and creates a new user. Is there anyway to get around this?

**Comment by s0h311** [comment]
Hi, @madhawk1296 try this:

In `auth.users` set `raw_app_meta_data` to

```Json
{
  "provider": "github",
  "providers": [
    "github"
  ]
}
```
and clear the`raw_user_meta_data`, so:
```Json
{}
```
In `auth.identities` clear the `identity_data`, so:
```Json
{}
```

and set the prodiver to `github`.

And then call `signInWithOAuth` setting github as provider. If you check the `created_at` and `id` in `auth.users` you'll see that they stayed the same. That's how I **changed the provider**. But in your case adding a `github` to the `providers` array  `raw_app_meta_data` column may do the trick.
